### ISSUE TITLE
Raise exception if attempting to connect to not existent actor

### ISSUE THREAD

**Issue by jackbond** [microsoft/service-fabric#984]
It would be extremely useful to have an overload for ActorProxy.Create<> that would raise an exception if the actor does not exist. As it stands, you have no way of determining if an actor exists except by connecting, attempting an operation, having the operation fail, and then actually having to delete the actor.

**Comment by gkhanna79** [comment]
@jackbond Can you please move this issue to https://github.com/Microsoft/service-fabric-services-and-actors-dotnet as we track all Actors issue there (see https://github.com/microsoft/service-fabric/blob/master/CONTRIBUTING.md)? @amanbha  can then triage.

**Comment by amanbha** [comment]
@jackbond Reliable Actors follow a virtual actor model in which an actor instance is not created explicitly, if the actor instance doesn't in service on a method call, its created. 
ActorProxy.Create is a local call and is not sent to the service, the call is actually made when a method is invoked for the actor, actors being virtual, if it doesn't exist in the service its created.
You can query the actors by using ActorServiceProxy.GetActorsAsync call and cache it, before making a call, you can check in this cached list.
Please feel free to open an issue in actors repo, if you have more question reg. this.

**Comment by jackbond** [comment]
I hadn't noticed this was closed. Using GetActorsAsync with some sort of cache isn't practical when a system hosts a large number of actors. Here's a use case, say I'm using an actor as a representation of a shopping cart, and I expose a controller method to retrieve the contents of a cart. If a user provides an id for a cart that doesn't exist, SF doesn't care, it goes ahead, creates an actor, and queries it. Now I've got a cart / data / overhead that I never wanted, and sadly, need to write a cleanup process to find. I could write some code to detect whether an actor has been created via the proper workflow, but it's pure overhead. Further, even if I detect it, I can't tell the actor to delete itself. In hindsight, I should have requested something like TryGetActor which simply returns null if the actor doesn't exist. Having to enumerate literally every actor, on every partition, and maintain a separate cache, to simply ascertain whether an actor is real is, quite frankly, nonsensical.
